x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/diff/calculation/algorithms/data_source_diff_algorithm.test.ts

```diff
@@ -421,81 +421,162 @@ describe('dataSourceDiffAlgorithm', () => {
+              conflict: ThreeWayDiffConflict.NONE,
+            })
+          );
+        });
+
+        it('if current version is undefined', () => {
+          const mockVersions: ThreeVersionsOf<RuleDataSource | undefined> = {
+            base_version: MissingVersion,
+            current_version: undefined,
+            target_version: {
+              type: DataSourceType.index_patterns,
+              index_patterns: ['one', 'three', 'four'],
+            },
+          };
+
+          const result = dataSourceDiffAlgorithm(mockVersions, false);
+
+          expect(result).toEqual(
+            expect.objectContaining({
+              has_base_version: false,
+              base_version: undefined,
+              merged_version: mockVersions.target_version,
+              diff_outcome: ThreeWayDiffOutcome.MissingBaseCanUpdate,
+              merge_outcome: ThreeWayMergeOutcome.Target,
+              conflict: ThreeWayDiffConflict.NONE,
+            })
+          );
+        });
+      });
+
+      describe('returns SOLVABLE conflict if rule is customized', () => {
+        it('if versions are different types', () => {
+          const mockVersions: ThreeVersionsOf<RuleDataSource | undefined> = {
+            base_version: MissingVersion,
+            current_version: { type: DataSourceType.data_view, data_view_id: '456' },
+            target_version: {
+              type: DataSourceType.index_patterns,
+              index_patterns: ['one', 'three', 'four'],
+            },
+          };
+
+          const result = dataSourceDiffAlgorithm(mockVersions, true);
+
+          expect(result).toEqual(
+            expect.objectContaining({
+              has_base_version: false,
+              base_version: undefined,
+              merged_version: mockVersions.target_version,
+              diff_outcome: ThreeWayDiffOutcome.MissingBaseCanUpdate,
+              merge_outcome: ThreeWayMergeOutcome.Target,
+              conflict: ThreeWayDiffConflict.SOLVABLE,
+            })
+          );
+        });
+
+        it('if current version is undefined', () => {
+          const mockVersions: ThreeVersionsOf<RuleDataSource | undefined> = {
+            base_version: MissingVersion,
+            current_version: undefined,
+            target_version: {
+              type: DataSourceType.index_patterns,
+              index_patterns: ['one', 'three', 'four'],
+            },
+          };
+
+          const result = dataSourceDiffAlgorithm(mockVersions, true);
+
+          expect(result).toEqual(
+            expect.objectContaining({
+              has_base_version: false,
+              base_version: undefined,
+              merged_version: mockVersions.target_version,
+              diff_outcome: ThreeWayDiffOutcome.MissingBaseCanUpdate,
+              merge_outcome: ThreeWayMergeOutcome.Target,
+              conflict: ThreeWayDiffConflict.SOLVABLE,
+            })
+          );
+        });
+      });
+    });
   });
```

Comment:
nit: Context and test names are reversed here. A general rule of thumb `describe` defines a context like `when rule is customized` or `with conflicts` and `it` says what's expected like `returns true` or `returns a SOLVABLE conflict`.